### ISSUE TITLE
Incorrect value return by GetCurrUTCDateTime() from codeunit 10 (Type Helper)

### ISSUE THREAD

**Issue by ftornero9** [Microsoft/AL#4625]
The procedure GetCurrUTCDateTime() from codeunit 10 (Type Helper) does not return the UTC DateTime

With this code the issue can be reproduced.

pageextension 50100 CustomerListExt extends "Customer List"
{
    trigger OnOpenPage();
    var
        cu10: Codeunit "Type Helper"; // 10
    begin
        Message('CurrentDateTime: %1\\\Codeunit Type Helper - GetCurrUTCDateTime %2\\\Codeunit Type Helper - GetCurrUTCDateTimeAsText %3',
                 CurrentDateTime(), cu10.GetCurrUTCDateTime(), cu10.GetCurrUTCDateTimeAsText());
    end;
}

This is the screenshot for the previous code, where the GetCurrUTCDateTimeAsText() procedure works fine but the GetCurrUTCDateTime() does not.

![image](https://user-images.githubusercontent.com/12720964/52975256-2000ad80-33c5-11e9-968e-a979c2d42048.png)

I did the test in this insider version:
![image](https://user-images.githubusercontent.com/12720964/52975362-7372fb80-33c5-11e9-9adf-b6f999eae178.png)

And in this sandbox one:

![image](https://user-images.githubusercontent.com/12720964/52975414-b7660080-33c5-11e9-9ee9-b86dcacc5416.png)

Regards.

**Comment by darjoo** [comment]
Hi @ftornero9 

CU10 GetCurrUTCDateTime does return UTC time, but when it is printed, like the code you have shared, it is implicitly converted to local time. As such this is by design and the expected result

**Comment by ftornero9** [comment]
Hello @darjoo, 

I don't think so, and int the other hand that would not be the expected result, at least not for me.

My machine is in GMT+1, you can see in the image the values.

format(DateTimeVar, 0, 9) give you the UTC date.

Assigning the funtion GetCurrUTCDateTime to a var don't give you the UTC date, but the GMT+1 date.

![image](https://user-images.githubusercontent.com/12720964/53100793-b26d9200-3528-11e9-9f8b-ebc00792b446.png)

**Comment by thpeder** [comment]
Hi @ftornero9,
I agree with you that it is not what you might expect, but this is not a new function and people might be relying on this behavior therefor it would be a breaking change to "fix" it.
The builtin Datetime type is limited in some ways but if we are hearing that it is big issues then we will improve it, but in the meantime if you are on-prem I would recommend using .net types instead if you need something with a higher level of control.

**Comment by ftornero9** [comment]
Hello @thpeder ,

I don't know if you have read my last post where I showed that the function is returning the current date ad time not the UTC one before it is displayed.

Are you saying that is the expected result too ?

Regards

**Comment by thpeder** [comment]
Hi @ftornero9,

Let me be completely honest with you, I might be missing something in how the system works with dates and times but my personal observation is that the data type is missing functionality but regarding the function there are most likely people relying on the unfortunate behaviour so I am very hesitant to say "we will fix it" because I'm not sure it can be fixed easily. 
I hope the text base version is good enough for you otherwise we need to look further into how we can improve the capabilities when it comes to date and time.

**Comment by ftornero9** [comment]
Hello @thpeder .

Actually I'm not using none of the functions in codeunit 10 dealing with dates.

I came across with GetCurrUTCDateTime recentely and after use it I got the issue I reported.

I don't belive that theres is many people relying in this function when don't return the UTC date/time but the same value that the internal function CurrentDateTime().

Anyway let the behaviour as is.

Regards.
